app/src/power.c

```diff
@@ -35,6 +36,20 @@ enum power_states sys_pm_policy_next_state(s32_t ticks) {
     s32_t current = k_uptime_get();
     if (power_last_uptime > 0 && !is_usb_power_present() &&
         current - power_last_uptime > MAX_IDLE_MS) {
+
+#if IS_ENABLED(CONFIG_ZMK_EXT_POWER)
```

Comment:
Instead of adding this here, look to switch the ext power driver to use `DEVICE_DEFINE` so you can pass a PM control function: https://docs.zephyrproject.org/2.3.0/reference/drivers/index.html#c.DEVICE_DEFINE